### PR TITLE
Get game when opening a donation

### PR DESCRIPTION
**Pull request by NateMoeller**
gamedropdown

### CODE REVIEW

File: api/src/main/java/com/nicknathanjustin/streamercontracts/donations/requests/CreateDonationRequest.java [hunk 1]
```diff
@@ -14,4 +15,5 @@
     @NonNull private String bounty;
     @NonNull private BigDecimal amount;
     @NonNull private String streamerUsername;
+    @Nullable private String game;
```
Comment (nckackerman) [inline api/src/main/java/com/nicknathanjustin/streamercontracts/donations/requests/CreateDonationRequest.java]: Whats the use case for a null game?

File: api/src/main/java/com/nicknathanjustin/streamercontracts/twitch/TwitchService.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+import java.util.List;
+import java.util.Map;
+
+public interface TwitchService {
+
+    /**
+     * Get the top games from the twitch api
+     *
+     * @return A List of top games
```
Comment (nckackerman) [inline api/src/main/java/com/nicknathanjustin/streamercontracts/twitch/TwitchService.java]: Does it? It looks like its returning a map of strings that map to a list of mapped object. Lets update the return to callout what each of the maps contain

File: web/src/app/donate/DonateComponent.js [hunk 3]
```diff
@@ -120,6 +122,27 @@ class DonateComponent extends Component {
     });
   }
 
+  getGameOptions() {
+    return this.props.topGames.map((game, idx) => ({
+      id: idx,
```
Comment (nckackerman) [inline web/src/app/donate/DonateComponent.js]: what is an idx?

File: web/src/app/donate/DonateComponent.js [hunk 4]
```diff
@@ -120,6 +122,27 @@ class DonateComponent extends Component {
...
+  }
+
+  renderOption(option, props, index) {
+    const width = 50;
+    const height = 75;
+    const url = option.img.replace('{width}', width).replace('{height}', height);
+
+    return (
+      <div>
+        <img className={styles.gameThumbnail} src={url} alt={option.name} width={width} height={height} />
```
Comment (nckackerman) [inline web/src/app/donate/DonateComponent.js]: I typically try and keep javascript free from css. What is the benefit for adding height and width directly to the image here and not adding it to the css className?

File: api/src/main/java/com/nicknathanjustin/streamercontracts/donations/requests/CreateDonationRequest.java (hunk 1 above)
Comment (nckackerman) [reply api/src/main/java/com/nicknathanjustin/streamercontracts/donations/requests/CreateDonationRequest.java]: Talked offline, null is needed for when type ahead doesnt find a match

File: web/src/app/donate/DonateComponent.js (hunk 3 above)
Comment (NateMoeller) [reply web/src/app/donate/DonateComponent.js]: JS shorthand for index :)

File: web/src/app/donate/DonateComponent.js (hunk 4 above)
Comment (NateMoeller) [reply web/src/app/donate/DonateComponent.js]: Width/Height here isn't css, its using the HTML width and height attributes. This is typically a better was to keep images the same size for layout purposes.

File: web/src/app/donate/DonateComponent.js (hunk 3 above)
Comment (nckackerman) [reply web/src/app/donate/DonateComponent.js]: ahh, cool. hadnt seen that before

**Pull request merged**